include/class.lock.php

```diff
@@ -146,5 +156,18 @@ static function cleanup() {
             'expire__lt' => SqlFunction::NOW()
         ))->delete();
     }
+
+    function validateToken($token=null, $code) {
+        if (!$code)
+            return false;
+
+        if (!($lock = Lock::lookup(array('code' => $code))))
+            return false;
+
+        if (strcmp($token, $lock->getToken()) !== 0)
+            return false;
+
+        return $lock->delete();
```

Comment:
Are you sure you want to delete the lock here -  What about if the action (e.g posReply) errors? Re-acquiring the lock will change the token (due to new expire date - please confirm) - so I don't the need to delete here. Lock cleaning mechanism should handle deleting expired sessions.